pilot/pkg/networking/core/v1alpha3/cluster.go

```diff
@@ -150,6 +151,16 @@ func normalizeClusters(push *model.PushContext, proxy *model.Proxy, clusters []*
 	return out
 }
 
+// castDestinationRuleOrDefault returns the destination rule enclosed by the config, if not null.
+// Otherwise, return defaul (empty) DR.
+func castDestinationRuleOrDefault(config *model.Config) *networking.DestinationRule {
+	if config != nil {
+		return config.Spec.(*networking.DestinationRule)
+	}
+
+	return &networking.DestinationRule{}
```

Comment:
if this is not mutated at all, is it possible to use a global constant here?

Reply:
Done